src/thor/triplegbuilder.cc

```diff
@@ -1297,7 +1310,7 @@ void TripLegBuilder::Build(
     float trim_end_pct = is_last_edge ? end_pct : 1;
 
     // Process the shape for edges where a route discontinuity occurs
-    uint32_t begin_index = (is_first_edge) ? 0 : trip_shape.size() - 1;
+    uint32_t begin_index = (is_first_edge || trip_shape.size() == 0) ? 0 : trip_shape.size() - 1;
```

Comment:
this seems unrelated please remove

Reply:
@kevinkreiser
We have an issue inside for (auto edge_itr = path_begin; edge_itr != path_end; ++edge_itr, ++edge_index) {
We can continue loop. And I think in case we skip first edge we can't use is_first_edge flag. And std::prev for iterator at all.
Maybe I should make throw instead of continue inside this loop?
Or will be better to make prev_edge_itr variable and use it instead of is_first_edge flag and std::prev(edge_itr)?

Reply:
About `trip_shape.size() == 0`. It can happen if we skip all iterations here

```
for (auto edge_itr = path_begin; edge_itr != path_end; ++edge_itr, ++edge_index) {
  const GraphId& edge = edge_itr->edgeid;
  graphtile = graphreader.GetGraphTile(edge, graphtile);
  if (graphtile == nullptr) {
    continue;
  }
  ```